src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -256,45 +250,39 @@ stepWithAxiom
     -- Combine the substitution produced by unification with the initial
     -- substitution carried by the configuration. Merging substitutions may
     -- produce another predicate during symbolic execution.
-    (     substitutionMergeCondition
-        , substitution
-        , _  -- TODO: Use this proof
-        ) <-
-            normalizeUnificationError
-                (makeFalsePredicate, [], EmptyUnificationProof)
-                existingVars
-                (mergeSubstitutions tools unificationSubstitution startSubstitution)
-
     normalizedSubstitutionWithCounter <-
-        normalizeSubstitutionError
-            (return PredicateSubstitution
-                { predicate = makeFalsePredicate
-                , substitution = []
-                }
+        stepperVariableToVariableForError
+            existingVars
+            $ unificationOrSubstitutionToStepError
```

Comment:
Note that having two different functions (`unificationOrSubstitutionToStepError` and `ctorSubstitutionCycleToBottom`) for error conversion solves the naming problem that the previous functions had (i.e. `substitutionToStepError` was misleading), but allows us to call one function without the other. I think that we never need to call the former without the latter, but doing so would produce hard to detect bugs, so I'd have preferred to either have this as a single function or to have an `unificationOrSubstitutionToStepError` which crashes when encountering a constructor cycle error.